### PR TITLE
Run3-sim118F Use token in getting collection and tfileservice for histograms in SimCalorimetry/EcalTrigPrimProducers

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Use token in getting collection and tfileservice for histograms in SimCalorimetry/EcalTrigPrimProducers

#### PR validation:

Use the runTheMatrix test workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimCalorimetry/EcalTrigPrimProducers/plugins/EcalTrigPrimAnalyzer.cc [hunk 1]
```diff
@@ -120,29 +99,26 @@ void EcalTrigPrimAnalyzer::analyze(const edm::Event &iEvent, const edm::EventSet
     return;
 
   // comparison with RecHits
-  edm::Handle<EcalRecHitCollection> rechit_EB_col;
-  iEvent.getByLabel(rechits_labelEB_, rechit_EB_col);
-
-  edm::Handle<EcalRecHitCollection> rechit_EE_col;
-  iEvent.getByLabel(rechits_labelEE_, rechit_EE_col);
+  const edm::Handle<EcalRecHitCollection> &rechit_EB_col = iEvent.getHandle(ebToken_);
+  const edm::Handle<EcalRecHitCollection> &rechit_EE_col = iEvent.getHandle(eeToken_);
```
Comment (thomreis) [inline SimCalorimetry/EcalTrigPrimProducers/plugins/EcalTrigPrimAnalyzer.cc:103]: The Handle is not really needed here as far as I can see. `Event::get` should work as well and is recommended.

File: SimCalorimetry/EcalTrigPrimProducers/plugins/EcalTrigPrimAnalyzer.cc [hunk 2]
```diff
@@ -120,29 +99,26 @@ void EcalTrigPrimAnalyzer::analyze(const edm::Event &iEvent, const edm::EventSet
...
-  edm::ESHandle<CaloSubdetectorGeometry> theBarrelGeometry_handle = iSetup.getHandle(barrelGeomToken_);
-  edm::ESHandle<CaloSubdetectorGeometry> theEndcapGeometry_handle = iSetup.getHandle(endcapGeomToken_);
+  const edm::ESHandle<CaloGeometry> &theGeometry = iSetup.getHandle(geomToken_);
+  const edm::ESHandle<CaloSubdetectorGeometry> &theBarrelGeometry_handle = iSetup.getHandle(barrelGeomToken_);
+  const edm::ESHandle<CaloSubdetectorGeometry> &theEndcapGeometry_handle = iSetup.getHandle(endcapGeomToken_);
 
   const CaloSubdetectorGeometry *theEndcapGeometry = theEndcapGeometry_handle.product();
   const CaloSubdetectorGeometry *theBarrelGeometry = theBarrelGeometry_handle.product();
-  edm::ESHandle<EcalTrigTowerConstituentsMap> eTTmap_ = iSetup.getHandle(eTTmapToken_);
+  const edm::ESHandle<EcalTrigTowerConstituentsMap> &eTTmap_ = iSetup.getHandle(eTTmapToken_);
```
Comment (thomreis) [inline SimCalorimetry/EcalTrigPrimProducers/plugins/EcalTrigPrimAnalyzer.cc:111]:
```suggestion
  const auto &eTTmap = iSetup.getData(eTTmapToken_);
```

File: SimCalorimetry/EcalTrigPrimProducers/plugins/EcalTrigPrimAnalyzer.cc [hunk 3]
```diff
@@ -59,48 +54,32 @@ EcalTrigPrimAnalyzer::EcalTrigPrimAnalyzer(const edm::ParameterSet &iConfig)
...
 
 // ------------ method called to analyze the data  ------------
 void EcalTrigPrimAnalyzer::analyze(const edm::Event &iEvent, const edm::EventSetup &iSetup) {
-  using namespace edm;
-  using namespace std;
-
   // Get input
-  edm::Handle<EcalTrigPrimDigiCollection> tp;
-  iEvent.getByLabel(label_, tp);
+  const edm::Handle<EcalTrigPrimDigiCollection> &tp = iEvent.getHandle(tpToken_);
```
Comment (thomreis) [inline SimCalorimetry/EcalTrigPrimProducers/plugins/EcalTrigPrimAnalyzer.cc:82]:
```suggestion
  const auto &tp = iEvent.get(tpToken_);
```
The Handle is not really needed here as far as I can see. Event::get should work as well and is recommended.

File: SimCalorimetry/EcalTrigPrimProducers/plugins/EcalTrigPrimAnalyzer.cc (hunk 1 above)
Comment (bsunanda) [reply SimCalorimetry/EcalTrigPrimProducers/plugins/EcalTrigPrimAnalyzer.cc]: done

File: SimCalorimetry/EcalTrigPrimProducers/plugins/EcalTrigPrimAnalyzer.cc [hunk 4]
```diff
@@ -120,29 +99,26 @@ void EcalTrigPrimAnalyzer::analyze(const edm::Event &iEvent, const edm::EventSet
...
-
-  edm::Handle<EcalRecHitCollection> rechit_EE_col;
-  iEvent.getByLabel(rechits_labelEE_, rechit_EE_col);
+  const edm::Handle<EcalRecHitCollection> &rechit_EB_col = iEvent.getHandle(ebToken_);
+  const edm::Handle<EcalRecHitCollection> &rechit_EE_col = iEvent.getHandle(eeToken_);
 
-  edm::ESHandle<CaloGeometry> theGeometry = iSetup.getHandle(geomToken_);
-  edm::ESHandle<CaloSubdetectorGeometry> theBarrelGeometry_handle = iSetup.getHandle(barrelGeomToken_);
-  edm::ESHandle<CaloSubdetectorGeometry> theEndcapGeometry_handle = iSetup.getHandle(endcapGeomToken_);
+  const edm::ESHandle<CaloGeometry> &theGeometry = iSetup.getHandle(geomToken_);
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply SimCalorimetry/EcalTrigPrimProducers/plugins/EcalTrigPrimAnalyzer.cc]: removed

File: SimCalorimetry/EcalTrigPrimProducers/plugins/EcalTrigPrimAnalyzer.cc [hunk 5]
```diff
@@ -120,29 +99,26 @@ void EcalTrigPrimAnalyzer::analyze(const edm::Event &iEvent, const edm::EventSet
...
+  const edm::ESHandle<CaloSubdetectorGeometry> &theBarrelGeometry_handle = iSetup.getHandle(barrelGeomToken_);
+  const edm::ESHandle<CaloSubdetectorGeometry> &theEndcapGeometry_handle = iSetup.getHandle(endcapGeomToken_);
 
   const CaloSubdetectorGeometry *theEndcapGeometry = theEndcapGeometry_handle.product();
   const CaloSubdetectorGeometry *theBarrelGeometry = theBarrelGeometry_handle.product();
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply SimCalorimetry/EcalTrigPrimProducers/plugins/EcalTrigPrimAnalyzer.cc]: taken care of

File: SimCalorimetry/EcalTrigPrimProducers/plugins/EcalTrigPrimAnalyzer.cc (hunk 3 above)
Comment (bsunanda) [reply SimCalorimetry/EcalTrigPrimProducers/plugins/EcalTrigPrimAnalyzer.cc]: done

**Comment by cecilecaillol** [comment]
+l1

**Comment by perrotta** [comment]
+1

**Pull request merged**